sdk/core/azure-core/src/main/java/com/azure/android/core/credential/TokenRequestHandle.java

```diff
@@ -0,0 +1,136 @@
+ * <p>
+ * It is important that an Event instance is processed only once by the {@link TokenRequestObserver}. The
+ * {@link TokenRequestObservable} uses a {@link androidx.lifecycle.LiveData} instance to post the Event to
+ * {@link TokenRequestObserver}. {@link androidx.lifecycle.LiveData} is designed to cache the latest value (in
+ * this case Event) and re-deliver it when there is a new Observer. An example for unexpected re-delivery is: Let's
+ * say the {@link TokenRequestObservable} is owned by a {@link androidx.lifecycle.ViewModel} which lives
+ * through multiple {@code Activity::onCreate} - {@code Activity::onDestroy} spans and in
+ * {@code Activity::onCreate} we register a {@link TokenRequestObserver} to this {@link androidx.lifecycle.ViewModel}
+ * 's scoped Observable. When a background thread requests for a token, the Observer will receive the request and
+ * process it. If the user begins rotating the screen (i.e configuration changed) then {@code Activity::onDestroy} is
+ * called and once the rotation is done {@code Activity::onCreate} method will be called again, which will result in
+ * the registration of a new {@link TokenRequestObserver} and will internally cause
+ * {@link androidx.lifecycle.LiveData} to re-deliver the old cached/handled Event, which we definitely don't want the
+ * new Observer to handle. To identify such an Event, the Observer uses {@link TokenRequestHandle#isConsumed()}.
+ */
+class TokenRequestHandle {
+    // The scope of the token requested by the handle owner.
+    private final List<String> scopes;
+    // Indicates whether the handle has already been consumed by a TokenRequestObserver.
+    private boolean isConsumed;
+    // The latch that signals the handle owner that the token acquisition operation is completed (successfully or
+    // error-ed).
+    private CountDownLatch latch = new CountDownLatch(1);
+    // The retrieved access token.
+    private AccessToken token;
+    // The error on token retrieval.
+    private Throwable error;
+
+    /**
+     * Creates a {@link TokenRequestHandle}.
+     *
+     * @param scopes The requested token scope.
+     */
+    public TokenRequestHandle(List<String> scopes) {
+        this.scopes = new ArrayList<>(scopes);
+    }
+
+    /**
+     * Synchronously wait for the arrival of an {@link AccessToken} requested via
+     * {@link TokenRequestObservable#sendRequest(List)}}.
+     *
+     * @param timeout The maximum time to wait in milliseconds.
+     * @return The {@link AccessToken}.
+     * @throws Throwable if there is an error.
+     */
+    public AccessToken waitForToken(Duration timeout) throws Throwable {
+        this.latch.await(timeout.toMillis(), TimeUnit.MILLISECONDS);
+
+        if (this.error != null) {
+            throw this.error;
+        } else {
+            return this.token;
+        }
+    }
+
+    /**
+     * PACKAGE PRIVATE METHOD.
```

Comment:
If these are package-private, why are we just using a comment rather than declaring them with package-private visibility (that is, no visibility modifier)? I know the class itself is package-private, but then why call out certain methods as package-private and others not?

Reply:
Good catch! This comes from the time these classes were originally written and I must have missed it when migrating them to azure-core.